signature/mechanism_gpgme.go

```diff
@@ -133,6 +134,30 @@ func (m *gpgmeSigningMechanism) Sign(input []byte, keyIdentity string) ([]byte,
 	if err != nil {
 		return nil, err
 	}
+
+	if path := os.Getenv(PASSPHRASE_FILE_ENVIRONMENT); path != "" {
+		logrus.Debugf("Reading user-specified passphrase for gpgme signing from %s=%s", PASSPHRASE_FILE_ENVIRONMENT, path)
+		buf, err := ioutil.ReadFile(path)
```

Comment:
Should this strip a trailing newline, so that `echo password > file` works? Only read the first line?

Probably the same behavior as `gpg2`.

Reply:
Very good though.  `gpg` reads the first line only, I'll change:

> --passphrase-file file
       Read  the  passphrase  from file file. Only the first line will be read from file file. This can only be used if only one
       passphrase is supplied. Obviously, a passphrase stored in a file is of questionable security if other users can read this
       file. Don't use this option if you can avoid it.

Reply:
Yeah, it’s not particularly pretty, but
- Careful users can use a Go API
- Humans who are likely to store the passphrase in files are, I think, fairly likely to use the same file for `gig` and for this.

I wouldn’t object to this being _stricter_ than `gpg` (e.g. rejecting files that contain any non-empty lines after the first one); I do think it makes sense for that first line to be, if possible, interpreted the same.